pkg/cableengine/ipsec/strongswan.go

```diff
@@ -291,15 +312,52 @@ func (i *strongSwan) loadSharedKey(endpoint types.SubmarinerEndpoint, client *go
 	return nil
 }
 
-func runCharon(debug bool, logFile string) error {
+const charonConfTemplate = `
+	charon {
+		port = {{.ipSecIKEPort}}
+		port_nat_t = {{.ipSecNATTPort}}
+		make_before_break = yes
+		ignore_acquire_ts = yes
+	}
+`
+
+func (i *strongSwan) writeCharonConfig(path string) {
+	err := os.Remove(path)
+	if err != nil {
+		klog.Infof("Error deleting %s", path)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		klog.Fatalf("Error creating %s: %s", path, err)
+	}
+	i.renderCharonConfigTemplate(f)
+}
+
+func (i *strongSwan) renderCharonConfigTemplate(f io.Writer) {
+	t, err := template.New("charon.conf").Parse(charonConfTemplate)
+	if err != nil {
+		klog.Fatalf("Error creating template for charon.conf")
+	}
+	if err = t.Execute(f, map[string]string{
+		"ipSecIKEPort":  i.ipSecIKEPort,
+		"ipSecNATTPort": i.ipSecNATTPort}); err != nil {
+		klog.Fatalf("Error rendering charon config file: %s", err)
+	}
+}
+
+func (i *strongSwan) runCharon() error {
+
+	i.writeCharonConfig("/etc/strongswan/strongswan.d/charon.conf")
```

Comment:
make this path a constant

Reply:
What's the benefit of that? :), it's good to have a constant when you'll use that string in several places. :-1:

Reply:
it's also good to have a constant when its value might need changing and you want to simplify that change and have it in a very visible part instead of buried down in the code. This particular path is tied to this particular package of strongswan. This might change if we change the base docker image, for example (which is what happened here btw)

Reply:
I will change it although I don't agree a lot with the usefulness , selecting a decent constant name won't do any harm either :)

I normally look for things like this with grep/editor search when I need, instead of looking at specific places in code.